client/components/migration/feature-announcement.jsx

```diff
@@ -0,0 +1,133 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { Flex, FlexItem, Modal, Icon, Button } from '@wordpress/components';
+import { useCallback, useState } from '@wordpress/element';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import bg from './images/wcshipping-migration.jpg';
+import { Dashboard, Preformatted, LessonPlan, Shipping } from './icons';
+import {
+	isEligableToMigrate,
+} from 'woocommerce/woocommerce-services/state/shipping-label/selectors';
+
+const FeatureAnnouncement = ({ translate, isEligable }) => {
+	const [isOpen, setIsOpen] = useState(isEligable);
+	const [isUpdating, setIsUpdating] = useState(false);
+
+	const closeModal = useCallback(() => {
```

Comment:
❔ – Curiosity – “Good to go, but I’d like to learn more.”
`useCallback` is used to optimize performance. To me, this feels like pre-mature optimization to skip re-rendering. Also, the 2nd parameter for dependencies is missing. How does React's `useCallback` if the dependency is empty? Does it always return the cache function or does it never return the cache function? How does it check when to use the cached version?

Reply:
When dependencies array is left empty or not provided, the function is always cached between renders., which is fine at least for now as these functions are simply a stub and are to be implemented properly when working on https://github.com/woocommerce/shipping/issues/123

Reply:
> When dependencies array is left empty or not provided, the function is always cached between renders.

I would like to learn more about this. Can you find me a reference please? It seems to contradict this https://react.dev/reference/react/useCallback#every-time-my-component-renders-usecallback-returns-a-different-function.

Reply:
> If you forget the dependency array, useCallback will return a new function every time:

I was wrong!
Would it work for you if we left this to the other PR that's gonna take care of the functionality of these buttons? Hopefully the implementer of that feature doesn't make the same mistake.

Reply:
No problem! Thanks for checking. Then, I suggest we remove `useCallback` for the `closeModal`, `snooze` and `update` functions to something like
```js
	const closeModal = () => setIsOpen(false);

	const snooze = () => {
		// Todo: implement maybe later
	};

	const update = () => {
		// Todo: implement update
		setIsUpdating(true);
		setTimeout(() => setIsUpdating(false), 2000);
	};
```

Whoever is working on these 3 functions can then decide to add the `useCallback` should there be a need for any performance optimization. Thoughts?

Reply:
>  should there be a need for any performance optimization

We don't always need to measure to find out about performance optimizations if:
1) We know there is a probable issue.
2) The so-to-say optimization comes with little to zero cost in terms of development time or readability, like in here.

Anyways I'm removing it for now, the other PR may add it though.